src/components/flowchart-wrapper/flowchart-wrapper.js

```diff
@@ -349,7 +350,7 @@ export const FlowChartWrapper = ({
           </div>
           {isRunningLocally() ? null : <ShareableUrlMetadata />}
         </div>
-        <ExportModal />
+        {displayExportBtn && <ExportModal />}
```

Comment:
do we need to do this, I am thinking if there's no exportBtn, there will be anyways no way for the user to change state.visible.exportModal = true so it will always be hidden.

Reply:
I agree, so my [comment](https://github.com/kedro-org/kedro-viz/pull/1983#discussion_r1684483742) is about that. It is better to check before the component being mounted. Like we do in other places of this file - `{displayMetadataPanel && <MetaData />}`

Reply:
As we agreed, there is no `state.visible.exportModal` https://github.com/kedro-org/kedro-viz/pull/1983/files#diff-8aec283ba10f2d8f1aa71cf147f0b016e3491e28087e437043e921d4956b1da9L35

 I made this `{displayExportBtn && <ExportModal />}` changes from Ravi's suggestion for 
```
 if (!display.exportBtn) {
    return null;
  }
```

Reply:
hi, there is state.visible.exportModal but by default it is false. It is mounted @ravi-kumar-pilla -- it's just the visibility is false.

Reply:
@rashidakanchwala , it is better we have it here rather than in the component

Reply:
okay, so my comment was to have checks outside of the component. It is just a foolproof check. As you said @rashidakanchwala we can definitely remove this completely. It is a good to have check. Earlier this was present inside the component

Reply:
I mean to say I completely remove `exportBtn`  from `visible` in our Redux state,

```
{
  visible: {
    code: false,
    exportBtn: true,
    exportModal: false,
    graph: true,
    labelBtn: true,
    layerBtn: true,
    metadataModal: false,
    miniMapBtn: true,
    modularPipelineFocusMode: null,
    expandPipelinesBtn: true,
    settingsModal: false,
    shareableUrlModal: false,
    sidebar: window.innerWidth > sidebarWidth.breakpoint,
  },
  display: {
    globalToolbar: true,
    sidebar: true,
    miniMap: true,
    expandAllPipelines: false,
  }
}
```

to

```
{
  visible: {
    code: false,
    graph: true,
    metadataModal: false,
    modularPipelineFocusMode: null,
    settingsModal: false,
    shareableUrlModal: false,
    miniMap: true,
    sidebar: window.innerWidth > sidebarWidth.breakpoint,
  },
  display: {
    globalToolbar: true,
    sidebar: true,
    miniMap: true,
    expandAllPipelines: false,
    metadataPanel: true,
    expandPipelinesBtn: true,
    exportBtn: true,
    labelBtn: true,
    layerBtn: true,
    exportModal: false,
    zoomToolbar: true,
  }  
}
```

Reply:
sounds good. thanks!

Reply:
@rashidakanchwala So sorry, there is `state.visible.exportModal` I got confused between  `exportModal` and `exportBtn`.